crates/typst-syntax/src/parser.rs

```diff
@@ -687,6 +687,60 @@ fn code_expr_prec(
 
             p.eat();
             code_expr_prec(p, false, prec, false);
+
+            // Special case for chained comparison operations.
+            if op.is_comp() && ast::BinOp::NotIn.precedence() >= min_prec {
+                if let Some(second_op) = ast::BinOp::from_kind_comp(p.current()) {
+                    let mut prec = second_op.precedence();
+                    if prec < min_prec {
+                        break;
+                    }
+
+                    match second_op.assoc() {
+                        ast::Assoc::Left => prec += 1,
+                        ast::Assoc::Right => {}
+                    }
+
+                    p.eat();
+                    code_expr_prec(p, false, prec, false);
+
+                    // Check whether it is a transitive comparison.
+                    match (op, second_op) {
+                        (ast::BinOp::Gt, ast::BinOp::Lt)
+                        | (ast::BinOp::Gt, ast::BinOp::Leq)
+                        | (ast::BinOp::Geq, ast::BinOp::Lt)
+                        | (ast::BinOp::Geq, ast::BinOp::Leq)
+                        | (ast::BinOp::Lt, ast::BinOp::Gt)
+                        | (ast::BinOp::Lt, ast::BinOp::Geq)
+                        | (ast::BinOp::Leq, ast::BinOp::Gt)
+                        | (ast::BinOp::Leq, ast::BinOp::Geq)
+                        | (ast::BinOp::Neq, ast::BinOp::Neq) => {
+                            if let Some(node) = p.node_mut(m) {
+                                node.convert_to_error(
+                                    "only transitive comparisons are allowed",
+                                );
+                            }
+                        }
+                        _ => {
+                            p.wrap(m, SyntaxKind::ChainedComp);
+                            continue;
```

Comment:
I think the wrapping should be the last thing that happens. This makes for clearer control flow.

Reply:
Ok, I think it should be fixed, please let me know. (once I have pushed the commit)